elastic/beats
# Make elasticsearch/node_stats metricset work for Stack Monitoring without xpack.enabled flag

## What does this PR do

This PR prepares the `elasticsearch/node_stats` metricset (without setting `xpack.enabled: true`) to collect and index data for the Stack Monitoring UI. Concretely, it does the following:
- introduces new fields, collecting data needed by the Stack Monitoring UI. These fields are named following [Metricbeat](https://www.elastic.co/guide/en/beats/devguide/current/event-conventions.html)/[ECS](https://www.elastic.co/guide/en/ecs/current/ecs-guidelines.html#_guidelines_for_field_names) naming conventions.
- introduces field aliases from current fields known to the Stack Monitoring UI to the fields defined by the metricset. This will aid transitioning the Stack Monitoring UI in 7.x to be driven off the metricset with `xpack.enabled` omitted. In 8.0 the aliases will be removed and the Stack Monitoring UI code will have to be overhauled to reference the new fields directly.
- removes the `data_xpack.go` file implementing the `xpack.enabled: true` code path for the `node_stats` metricset. Going forward, the `node_stats` metricset will simply ignore the value of the `xpack.enabled` setting.

## Testing this PR

1. Build Metricbeat with this PR checked out.
   ```
   cd $GOPATH/src/github.com/elastic/beats/metricbeat
   mage build
   ```

2. Enable the `elasticsearch-xpack` module.
   ```
   metricbeat modules enable elasticsearch-xpack
   ```

3. Optionally, disable the `system` module. This will reduce "noise" in `metricbeat-*` indices, should you needs to inspect them.
   ```
   metricbeat modules disable system
   ```

3. Run Metricbeat (assumes output is being sent to Elasticsearch running at localhost:9200).
   ```
   metricbeat -e
   ```

## Authors checklist
* [x] Define new fields in `elasticsearch/node_stats/_meta/fields.yml`
* [x] Enhance `data.go` to collect necessary additional metrics and index them into new fields.
* [x] Define field aliases from custom field names to Metricbeat/ECS-compliant field names in `elasticsearch/_meta/fields.yml`
